src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -91,10 +91,12 @@ - (BOOL)isConnecting;
         dispatch_queue_t bleWorkQueue;
 
         void BleConnectionDelegateImpl::NewConnection(
-            Ble::BleLayer * bleLayer, void * appState, const SetupDiscriminator & deviceDiscriminator)
+            Ble::BleLayer * bleLayer, void * appState, const SetupDiscriminator & inDeviceDiscriminator)
         {
             assertChipStackLockedByCurrentThread();
 
+            SetupDiscriminator deviceDiscriminator = inDeviceDiscriminator;
```

Comment:
This is still on the stack right? So if the caller is invalid, the staack of this method seems invalid as well.

To me it sounds like we need some capture-by-value or heap-allocation to be freed by the dispatch instead.

Reply:
Blocks do capture-by-value.  The problem is that currently what's being captured by value is a reference, not an actual object....  This is the right fix.